packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/DeleteCategoriesTests.java

```diff
@@ -0,0 +1,102 @@
+import static org.springframework.http.HttpStatus.FORBIDDEN;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+
+public class DeleteCategoriesTests extends RestTest {
+
+    private UserModel user;
+
+
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation() throws Exception
+    {
+        STEP("Create a user");
+        user = dataUser.createRandomTestUser();
+    }
+
+    /**
+     * Check we can delete a category.
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteCategory()
+    {
+        RestCategoryModel aCategory = createCategory();
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingCategory(aCategory).deleteCategory();
+        restClient.assertStatusCodeIs(NO_CONTENT);
+    }
+
+    /**
+     * Check we get an error when trying to delete a category as a non-admin user.
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteCategoryAsRegularUser_andFail()
+    {
+        RestCategoryModel aCategory = createCategory();
+        restClient.authenticateUser(user).withCoreAPI().usingCategory(aCategory).deleteCategory();
+        restClient.assertStatusCodeIs(FORBIDDEN).assertLastError().containsSummary("Current user does not have permission to delete a category");
+    }
+
+    /**
+     * Check we receive 404 error when trying to delete a category with a non-existent node id.
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteNonExistentCategory()
+    {
+        STEP("Get category with non-existent id");
+        final RestCategoryModel rootCategory = new RestCategoryModel();
+        final String id = "non-existing-dummy-id";
+        rootCategory.setId(id);
+
+        STEP("Attempt to delete category with non-existent id and receive 404");
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingCategory(rootCategory).deleteCategory();
+        restClient.assertStatusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Attempt to delete a category when providing a node id that doesn't belong to a category
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteCategory_givenNonCategoryNodeId()
+    {
+        STEP("Create a site and a folder inside it");
+        final SiteModel site = dataSite.usingUser(user).createPublicRandomSite();
+        final FolderModel folder = dataContent.usingUser(user).usingSite(site).createFolder();
+        String id = folder.getNodeRef();
+
+        final RestCategoryModel aCategory = new RestCategoryModel();
+        aCategory.setId(id);
+
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingCategory(aCategory).deleteCategory();
+        restClient.assertStatusCodeIs(BAD_REQUEST).assertLastError().containsSummary("Node id does not refer to a valid category");
+    }
+
+    public RestCategoryModel createCategory()
```

Comment:
I think Krys created a helper method for this. No problem if you don't want to use it, but just in case you wanted to reduce the code.